### PR TITLE
replace babel require hook with a build step

### PR DESCRIPTION
**Pull request by joeybaker**
The babel require hook is explicitly not suitable for libraries:
https://babeljs.io/docs/usage/require/ 

I’ve noticed that in trying to use this module, and trying to use the 
babel require hook as well, babel stops working. This patch removes the
require hook, adds a build process, and runs that build process before
a `npm publish` and `npm test`

### CODE REVIEW

**Comment by sullenor** [comment]
Thank you for your help :)
I put it as a temporary solution for projects, which doesn't use babel, because I was uncertain how to compile it.

**Comment by joeybaker** [comment]
Of course! Thanks for building this!

**Comment by joeybaker** [comment]
FYI: there are failing tests in master right now. I was a bit unsure if that was intentional?

File: index.js [hunk 1]
```diff
@@ -2,18 +2,4 @@
...
-    return '^' + escape(path.join(__dirname, i) + path.sep);
-}).join('|');
-
-require('babel/register')({
-    only:   new RegExp('(' + regexp + ')'),
-    ignore: false,
-    loose: 'all'
-});
-
 module.exports = require('./src');
```
Comment (sullenor) [inline index.js]: Is it necessary to fix path to the dir here? Looks like we will get `dist` after compile?

Comment (joeybaker) [inline index.js]: whoops, you're right. That slipped through, my bad!

Comment (joeybaker) [inline index.js]: fixed!

**Comment by sullenor** [comment]
Since I found out that I have same goals here as in `css-modules-loader-core`, I decided to copy all test-cases from that lib and run `css-modules-require-hook` through them. I managed to fix few, but some of them still broken. So its ok :)

**Comment by sullenor** [comment]
Looks like its ok now, thank you for your help again :)

**Pull request merged**